src/Tizen.Peripheral/Tizen.Peripheral/Gpio.cs

```diff
@@ -0,0 +1,233 @@
+            _handle = handle;
+            PinNumber = pinNumber;
+        }
+
+        /// <summary>
+        /// Opens a GPIO pin.
+        /// </summary>
+        /// <param name="pinNumber">The GPIO pin number.</param>
+        /// <param name="mode">GPIO direction.</param>
+        /// <returns>GPIO object.</returns>
+        public static Gpio Open(int pinNumber, Mode mode)
+        {
+            var ret = NativeGpio.Open(pinNumber, out IntPtr handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            var gpio = new Gpio(handle, pinNumber);
+
+            switch (mode)
+            {
+                case Mode.Input:
+                    ret = NativeGpio.SetEdgeMode(gpio._handle, NativeGpio.EdgeType.Both);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.In);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    gpio.SetIntteruptedCallback();
+                    gpio._interruptedEventCallback(handle, ErrorCode.None, IntPtr.Zero);
+                    break;
+                case Mode.OutputInitiallyLow:
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutLow);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    break;
+                case Mode.OutputInitiallyHigh:
+                    ret = NativeGpio.SetDirection(handle, NativeGpio.Direction.OutHigh);
+                    if (ret != ErrorCode.None)
+                        throw ExceptionFactory.CreateException(ret);
+
+                    break;
+            }
+
+            return gpio;
+        }
+
+        private void SetIntteruptedCallback()
+        {
+            _interruptedEventCallback = OnInterrupted;
+            var ret = NativeGpio.SetInterruptedCb(_handle, _interruptedEventCallback, IntPtr.Zero);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        private void OnInterrupted(IntPtr handle, ErrorCode error, IntPtr data)
+        {
+            PinUpdated?.Invoke(this, new PinUpdatedEventArgs(this, PinNumber, PinValue));
+        }
+
+        /// <summary>
+        /// Closes a GPIO pin.
+        /// </summary>
+        public void Close()
+        {
+            var ret = NativeGpio.UnsetInterruptedCb(_handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            ret = NativeGpio.Close(_handle);
+            if (ret != ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Disposes GPIO.
+        /// </summary>
+        public void Dispose()
```

Comment:
Implement the dispose pattern
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose